src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -46,46 +44,95 @@ internal static BlobBuilder MethodSignatureEncoder(ModuleBuilderImpl _module, Ty
-                    signature.Int64();
-                    break;
-                case "System.Object":
-                    signature.Object();
-                    break;
-                case "System.String":
-                    signature.String();
-                    break;
-
-                default: throw new NotSupportedException(SR.Format(SR.NotSupported_Signature, type.FullName));
+                switch (typeId)
+                {
+                    case CoreTypeId.Boolean:
+                        signature.Boolean();
+                        return;
+                    case CoreTypeId.Byte:
+                        signature.Byte();
+                        return;
+                    case CoreTypeId.SByte:
+                        signature.SByte();
+                        return;
+                    case CoreTypeId.Char:
+                        signature.Char();
+                        return;
+                    case CoreTypeId.Int16:
+                        signature.Int16();
+                        break;
+                    case CoreTypeId.UInt16:
+                        signature.UInt16();
+                        return;
+                    case CoreTypeId.Int32:
+                        signature.Int32();
+                        return;
+                    case CoreTypeId.UInt32:
+                        signature.UInt32();
+                        return;
+                    case CoreTypeId.Int64:
+                        signature.Int64();
+                        return;
+                    case CoreTypeId.UInt64:
+                        signature.UInt64();
+                        return;
+                    case CoreTypeId.Single:
+                        signature.Single();
+                        return;
+                    case CoreTypeId.Double:
+                        signature.Double();
+                        return;
+                    case CoreTypeId.IntPtr:
+                        signature.IntPtr();
+                        return;
+                    case CoreTypeId.UIntPtr:
+                        signature.UIntPtr();
+                        return;
+                    case CoreTypeId.Object:
+                        signature.Object();
+                        return;
+                    case CoreTypeId.String:
+                        signature.String();
+                        return;
+                }
             }
+
+            throw new NotSupportedException(SR.Format(SR.NotSupported_Signature, type.FullName));
         }
     }
+
+    internal enum CoreTypeId
+    {
+        Void = 0,
+        Object = 1,
+        Boolean = 2,
+        Char = 3,
+        SByte = 4,
+        Byte = 5,
+        Int16 = 6,
+        UInt16 = 7,
+        Int32 = 8,
+        UInt32 = 9,
+        Int64 = 10,
+        UInt64 = 11,
+        Single = 12,
+        Double = 13,
+        Decimal = 14,
+        DateTime = 15,
+        String = 16,
+        IntPtr = 17,
+        UIntPtr = 18,
```

Comment:
```suggestion
        Void,
        Object,
        Boolean,
        Char,
        SByte,
        Byte,
        Int16,
        UInt16,
        Int32,
        UInt32,
        Int64,
        UInt64,
        Single,
        Double,
        String,
        IntPtr,
        UIntPtr,
```
No need to manually assign values to internal enum members. The compiler will do it for you.